canonical/kfp-operators
# Updated access and secret key handling [KF-208] and added more integration tests [KF-212]

1. Modified access and secret key handling to pass them as secrets instead of environment variables, therefore not exposing them in plain text [KF-208]
2. Expanded the integration tests for kfp-profile-controller [KF-212]
- Deploy kubeflow-profiles and metacontroller-operator required for CRDs creation
- Test if the charms are deployed and active
- Create a test profile and corresponding namespace with label pipelines.kubeflow.org/enabled=true
- Ensure that the default-editor service account was created
- Check whether the secret exists and contains the expected key
- Ensure that the pods are ready
3. Modified charm.py due to black lint formatting
4. Added juju agent version in bootstrap options (NOTE: to be removed once juju is fixed, see [KF-231])

[KF-208]: https://warthogs.atlassian.net/browse/KF-208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[KF-212]: https://warthogs.atlassian.net/browse/KF-212?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[KF-231]: https://warthogs.atlassian.net/browse/KF-231
